src/mavsdk/plugins/camera/camera_definition.cpp

```diff
@@ -203,6 +206,28 @@ bool CameraDefinition::parse_xml()
                 continue;
             }
             new_parameter->options = maybe_options.second;
+            
+            auto maybe_default = find_default(new_parameter->options, default_str);
+            
+            if (!maybe_default.first) {
+                LogWarn() << "Default not found for " << param_name;
+                return false;
+            }
+            
+            new_parameter->default_option = maybe_default.second;
```

Comment:
Isn't that copy-pasted from below? Could we put that into a separate function?

Reply:
I actually had that done via lambda, but it didn't save much code. 

The lambda:
```cpp
auto get_default_opt = [&](){
          auto maybe_default = find_default(new_parameter->options, default_str);

          if (!maybe_default.first) {
              LogWarn() << "Default not found for " << param_name;
              return std::optional<Option>{};
          }

          return std::optional{maybe_default.second};
      };
```

Usages:
```cpp
            if (auto default_option = get_default_opt()) {
                new_parameter->default_option = default_option;
            } else {
                return false;
            }
```
The main abstraction "issue" is the `return false;` without it you wouldn't need to check the result of a function.

If you feel that that is a better way to abstract that I'll change it. :)

Reply:
I think it's better! It makes an implementation detail of what is inside `get_default_opt()`, and it avoids some duplication (like the log message).

Reply:
Okay! Thx for the feedback, I've used the lambda.